components/opponent/wikis/warcaft/opponent_display_custom.lua

```diff
@@ -0,0 +1,109 @@
+---
+-- @Liquipedia
+-- wiki=warcraft
+-- page=Module:OpponentDisplay/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local DisplayUtil = require('Module:DisplayUtil')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local TypeUtil = require('Module:TypeUtil')
+
+local Opponent = Lua.import('Module:Opponent/Custom', {requireDevIfEnabled = true})
+local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+local MatchGroupUtil = Lua.import('Module:MatchGroup/Util/Custom', {requireDevIfEnabled = true})
+local PlayerDisplay = Lua.import('Module:Player/Display/Custom', {requireDevIfEnabled = true})
+
+local CustomOpponentDisplay = {propTypes = {}, types={}}
+
+CustomOpponentDisplay.propTypes.InlineOpponent = {
+	flip = 'boolean?',
+	opponent = MatchGroupUtil.types.GameOpponent,
+	showFlag = 'boolean?',
+	showLink = 'boolean?',
+	showRace = 'boolean?',
+	teamStyle = TypeUtil.optional(OpponentDisplay.types.TeamStyle),
+}
+
+function CustomOpponentDisplay.InlineOpponent(props)
+	DisplayUtil.assertPropTypes(props, CustomOpponentDisplay.propTypes.InlineOpponent)
+	local opponent = props.opponent
+
+	if Opponent.partySize((opponent or {}).type) then -- opponent.type == 'solo' 'duo' 'trio' 'quad'
+		return CustomOpponentDisplay.InlinePlayers(props)
+	end
+
+	return OpponentDisplay.InlineOpponent(props)
+end
+
+CustomOpponentDisplay.propTypes.BlockOpponent = {
+	flip = 'boolean?',
+	opponent = MatchGroupUtil.types.GameOpponent,
+	overflow = TypeUtil.optional(DisplayUtil.types.OverflowModes),
+	showFlag = 'boolean?',
+	showLink = 'boolean?',
+	showPlayerTeam = 'boolean?',
+	showRace = 'boolean?',
+	teamStyle = TypeUtil.optional(OpponentDisplay.types.TeamStyle),
+	playerClass = 'string?',
+	abbreviateTbd = 'boolean?',
+}
+
+function CustomOpponentDisplay.BlockOpponent(props)
+	DisplayUtil.assertPropTypes(props, CustomOpponentDisplay.propTypes.BlockOpponent)
+	local opponent = props.opponent
+
+	opponent.extradata = opponent.extradata or {}
+	-- Default TBDs to not show links
+	local showLink = Logic.nilOr(props.showLink, not Opponent.isTbd(opponent))
+
+	if opponent.type == 'literal' and opponent.extradata.hasRaceOrFlag then
+		props.showRace = false
+		return CustomOpponentDisplay.BlockPlayers(Table.merge(props, {showLink = showLink}))
```

Comment:
Why is literal calling BlockPlayers?

Reply:
because if `opponent.extradata.hasRaceOrFlag` is true then we have a player opponent that is not yet determined (TBD) for which we already now race or flag, hence need to use player display component not literal (literal does not show flag/race)

Reply:
Is this something WC does right now?

Reply:
they do